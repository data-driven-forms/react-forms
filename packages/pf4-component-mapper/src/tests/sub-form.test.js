import React from 'react';
import { mount } from 'enzyme';

import FormRenderer, { componentTypes } from '@data-driven-forms/react-form-renderer';
import FormTemplate from '../files/form-template';
import componentMapper from '../files/component-mapper';
import { Title, Text } from '@patternfly/react-core';

describe('SubForm component', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      componentMapper,
      FormTemplate,
      onSubmit: jest.fn()
    };
  });

  it('should render SubForm correctly', () => {
    const schema = {
      fields: [
        {
          component: componentTypes.SUB_FORM,
          name: 'subform',
          title: 'some title',
          description: 'some description',
          fields: [
            {
              component: componentTypes.TEXT_FIELD,
              name: 'some input'
            }
          ]
        }
      ]
    };
    const wrapper = mount(<FormRenderer {...initialProps} schema={schema} />);

    expect(wrapper.find(Title).text()).toEqual('some title');
    expect(wrapper.find(Text).text()).toEqual('some description');
    expect(wrapper.find('input')).toHaveLength(1);
  });

  it('should render SubForm with description correctly', () => {
    const schema = {
      fields: [
        {
          component: componentTypes.SUB_FORM,
          name: 'subform',
          title: 'some title',
          fields: [
            {
              component: componentTypes.TEXT_FIELD,
              name: 'some input'
            }
          ]
        }
      ]
    };
    const wrapper = mount(<FormRenderer {...initialProps} schema={schema} />);

    expect(wrapper.find(Title).text()).toEqual('some title');
    expect(wrapper.find(Text)).toHaveLength(0);
    expect(wrapper.find('input')).toHaveLength(1);
  });

  it('should render SubForm without title correctly', () => {
    const schema = {
      fields: [
        {
          component: componentTypes.SUB_FORM,
          name: 'subform',
          description: 'some description',
          fields: [
            {
              component: componentTypes.TEXT_FIELD,
              name: 'some input'
            }
          ]
        }
      ]
    };
    const wrapper = mount(<FormRenderer {...initialProps} schema={schema} />);

    expect(wrapper.find(Title)).toHaveLength(0);
    expect(wrapper.find(Text).text()).toEqual('some description');
    expect(wrapper.find('input')).toHaveLength(1);
  });
});
