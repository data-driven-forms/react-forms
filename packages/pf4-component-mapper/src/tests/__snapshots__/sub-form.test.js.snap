// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubForm component should render SubForm correctly 1`] = `
<Fragment>
  <Title
    size="xl"
  >
    cosiTitle
  </Title>
  <div>
    Here would be form
  </div>
</Fragment>
`;

exports[`SubForm component should render SubForm with description correctly 1`] = `
<Fragment>
  <Title
    size="xl"
  >
    cosiTitle
  </Title>
  <TextContent>
    <Text
      component="small"
      style={
        Object {
          "marginBottom": 0,
        }
      }
    >
      description here!
    </Text>
  </TextContent>
  <div>
    Here would be form
  </div>
</Fragment>
`;

exports[`SubForm component should render SubForm without title correctly 1`] = `
<Fragment>
  <div>
    Here would be form
  </div>
</Fragment>
`;
